code/datums/martial/_martial.dm

```diff
@@ -34,12 +34,14 @@
 	if(length(streak) > max_streak_length)
 		streak = copytext(streak, 1 + length(streak[1]))
 	if (display_combos)
-		holder.hud_used.combo_display.update_icon_state(streak)
+		var/mob/living/holder_living = holder.resolve()
+		holder_living.hud_used.combo_display.update_icon_state(streak)
```

Comment:
```suggestion
		holder_living?.hud_used?.combo_display?.update_icon_state(streak)
```

The thing to know about weakrefs is `resolve` can return null if the object was deleted.

Reply:
OK should it just return if the holder no longer exists?